File: com/RaghavStaadExtractor/ExecutionValidation.vb
Imports System
Imports System.IO
Imports System.Management
Imports System.Runtime.InteropServices
Imports System.Security.Cryptography
Imports System.Text
Imports Microsoft.VisualBasic

<ComVisible(False)>
Friend Class ExecutionValidation

    Friend Class LicenseCheckResult
        Friend Property Status As LicenseStatus
        Friend Property ExpiryDate As String
    End Class

    Friend Enum LicenseStatus
        Valid
        Expired
        Unauthorized
        InvalidFormat
    End Enum

    Private ReadOnly aesKey As String
    Private ReadOnly aesIV As String
    Private ReadOnly fakeErrors As String()

    Public Sub New()
        aesKey = "AshuVedantRaghav"
        aesIV = "AshuVedantRaghav"
        fakeErrors = New String() {
            "Runtime Error 0x80070005: Access is denied.",
            "Unhandled Exception: System.IO.FileLoadException",
            "Fatal Error: Missing dependency 'mscorlib.dll'.",
            "Memory read violation at address 0x00000014.",
            "System.NullReferenceException at MainModule()",
            "Unexpected token in config file at line 42.",
            "Access violation while reading registry key.",
            "System.BadImageFormatException: Invalid executable format.",
            "Fatal: CLR runtime initialization failed.",
            "Exception code 0xc0000005: Access violation."
        }
    End Sub

    ' Validates the trial/license state. The C:\PlateNova\PlateNova.tst cache file stores the
    ' timestamp of the last successful check; if the system clock is ever behind that stored
    ' timestamp (a classic trial-reset trick), validation is rejected as tampering.
    Friend Function IsLicenceValid() As Boolean
        Try
            Dim now As DateTime = DateTime.Now
            Dim path As String = "C:\PlateNova\PlateNova.tst"

            If File.Exists(path) Then
                Dim encryptedBase As String = File.ReadAllText(path)
                Dim decrypted As String = AES_Decrypt(encryptedBase)
                Dim cachedDate As DateTime
                If Not DateTime.TryParse(decrypted, cachedDate) Then
                    ShowFakeError()
                    Return False
                ElseIf DateTime.Compare(now, cachedDate) < 0 Then
                    ' Clock moved backwards since the last successful check - treat as tampering
                    ShowFakeError()
                    Return False
                End If
                ' now >= cachedDate: fall through to the real validation below
            End If

            Dim biosSerial As String = GetBiosSerial().ToUpper()
            If String.IsNullOrWhiteSpace(biosSerial) Then
                ShowFakeError()
                Return False
            End If

            Dim isValid As Boolean
            Dim trialCutoff As New DateTime(2025, 12, 31)
            If DateTime.Compare(DateTime.Today, trialCutoff) <= 0 Then
                ' Within the free trial window - no license file required yet
                isValid = True
            Else
                Dim licPath As String = "C:\PlateNova\ExecutionFile.lic"
                If Not File.Exists(licPath) Then
                    ShowFakeError()
                    Return False
                End If

                Dim checkResult As LicenseCheckResult = ParseLicenseFile(licPath, biosSerial)
                If checkResult.Status <> LicenseStatus.Valid Then
                    ShowFakeError()
                    Return False
                End If
                isValid = True
            End If

            If isValid Then
                Try
                    Dim plainText As String = now.ToString("yyyy-MM-dd HH:mm:ss")
                    Dim encryptedNow As String = AES_Encrypt(plainText)
                    If Not String.IsNullOrEmpty(encryptedNow) Then
                        File.WriteAllText(path, encryptedNow)
                    End If
                Catch
                    ' Ignore cache-write failures; they don't affect the validity result
                End Try
            End If

            Return isValid

        Catch ex As Exception
            ShowFakeError()
            Return False
        End Try
    End Function

    ' Each line of the license file is "<BIOS-SERIAL>=<expiry-date | forever | permanent>",
    ' individually AES-encrypted. Returns the first line matching biosSerial, else Unauthorized.
    Private Function ParseLicenseFile(licPath As String, biosSerial As String) As LicenseCheckResult
        Try
            For Each rawLine As String In File.ReadAllLines(licPath)
                Dim decrypted As String = AES_Decrypt(rawLine).Trim()
                If String.IsNullOrWhiteSpace(decrypted) OrElse decrypted.StartsWith("#") OrElse
                   decrypted.StartsWith("//") OrElse Not decrypted.Contains("=") Then
                    Continue For
                End If

                Dim parts As String() = decrypted.Split("="c)
                If parts.Length <> 2 Then Continue For

                Dim key As String = parts(0).Trim().ToUpper()
                Dim value As String = parts(1).Trim().ToLower()

                If biosSerial = key Then
                    If value = "forever" OrElse value = "permanent" Then
                        Return New LicenseCheckResult With {.Status = LicenseStatus.Valid, .ExpiryDate = "Forever"}
                    End If

                    Dim expiry As DateTime
                    If Not DateTime.TryParse(value, expiry) Then
                        Return New LicenseCheckResult With {.Status = LicenseStatus.InvalidFormat, .ExpiryDate = value}
                    ElseIf DateTime.Compare(DateTime.Today, expiry) > 0 Then
                        Return New LicenseCheckResult With {.Status = LicenseStatus.Expired, .ExpiryDate = expiry.ToString("yyyy-MM-dd")}
                    Else
                        Return New LicenseCheckResult With {.Status = LicenseStatus.Valid, .ExpiryDate = expiry.ToString("yyyy-MM-dd")}
                    End If
                End If
            Next

            Return New LicenseCheckResult With {.Status = LicenseStatus.Unauthorized, .ExpiryDate = ""}

        Catch ex As Exception
            Return New LicenseCheckResult With {.Status = LicenseStatus.InvalidFormat, .ExpiryDate = ex.Message}
        End Try
    End Function

    Private Function AES_Decrypt(encryptedBase64 As String) As String
        Try
            Dim keyBytes As Byte() = Encoding.UTF8.GetBytes(aesKey)
            Dim ivBytes As Byte() = Encoding.UTF8.GetBytes(aesIV)
            Dim buffer As Byte() = Convert.FromBase64String(encryptedBase64)

            Using aes As Aes = Aes.Create()
                aes.Key = keyBytes
                aes.IV = ivBytes
                aes.Padding = PaddingMode.PKCS7
                Using transform As ICryptoTransform = aes.CreateDecryptor()
                    Using stream As New MemoryStream(buffer)
                        Using cryptoStream As New CryptoStream(stream, transform, CryptoStreamMode.Read)
                            Using reader As New StreamReader(cryptoStream)
                                Return reader.ReadToEnd()
                            End Using
                        End Using
                    End Using
                End Using
            End Using
        Catch ex As Exception
            Return ""
        End Try
    End Function

    Private Function AES_Encrypt(plainText As String) As String
        Try
            Dim keyBytes As Byte() = Encoding.UTF8.GetBytes(aesKey)
            Dim ivBytes As Byte() = Encoding.UTF8.GetBytes(aesIV)

            Using aes As Aes = Aes.Create()
                aes.Key = keyBytes
                aes.IV = ivBytes
                aes.Padding = PaddingMode.PKCS7
                Using transform As ICryptoTransform = aes.CreateEncryptor()
                    Using memoryStream As New MemoryStream()
                        Using cryptoStream As New CryptoStream(memoryStream, transform, CryptoStreamMode.Write)
                            Using writer As New StreamWriter(cryptoStream)
                                writer.Write(plainText)
                            End Using
                        End Using
                        Return Convert.ToBase64String(memoryStream.ToArray())
                    End Using
                End Using
            End Using
        Catch ex As Exception
            Return ""
        End Try
    End Function

    Private Function GetBiosSerial() As String
        Try
            Dim searcher As New ManagementObjectSearcher("SELECT SerialNumber FROM Win32_BIOS")
            For Each item As ManagementObject In searcher.Get()
                Dim serial As String = item("SerialNumber").ToString().Trim()
                If Not String.IsNullOrWhiteSpace(serial) Then
                    Return serial
                End If
            Next
        Catch ex As Exception
            ' Ignore and fall through to empty result
        End Try
        Return ""
    End Function

    ' Shows a random fake system-error message (a deterrent, not a real crash) and, best-effort,
    ' closes the active workbook without saving so the user loses no work but can't continue.
    Private Sub ShowFakeError()
        Try
            Dim rnd As New Random()
            Dim prompt As String = fakeErrors(rnd.Next(fakeErrors.Length))
            Interaction.MsgBox(prompt, MsgBoxStyle.Critical, "Fatal Error")

            Try
                Dim excelApp As Object = Nothing
                Try
                    excelApp = Marshal.GetActiveObject("Excel.Application")
                Catch
                    ' Excel isn't running - nothing more to do
                End Try

                If excelApp IsNot Nothing Then
                    Dim activeWorkbook As Object = excelApp.ActiveWorkbook
                    If activeWorkbook IsNot Nothing Then
                        activeWorkbook.Close(False)
                    End If
                    Marshal.ReleaseComObject(excelApp)
                End If
            Catch
                ' Ignore cleanup errors
            End Try
        Catch
            ' This is a best-effort deterrent and must never throw
        End Try
    End Sub

    Friend Function GetCurrentBiosSerial() As String
        Return GetBiosSerial()
    End Function

    Friend Function ValidateLicenseFileFormat(licPath As String) As Boolean
        Try
            If Not File.Exists(licPath) Then Return False

            For Each rawLine As String In File.ReadAllLines(licPath)
                Dim decrypted As String = AES_Decrypt(rawLine).Trim()
                If String.IsNullOrWhiteSpace(decrypted) OrElse decrypted.StartsWith("#") OrElse Not decrypted.Contains("=") Then
                    Continue For
                End If

                Dim parts As String() = decrypted.Split("="c)
                If parts.Length = 2 AndAlso Not String.IsNullOrWhiteSpace(parts(0)) AndAlso Not String.IsNullOrWhiteSpace(parts(1)) Then
                    Return True
                End If
            Next

            Return False
        Catch ex As Exception
            Return False
        End Try
    End Function

End Class

File: com/RaghavStaadExtractor_decompiled_fresh/ExecutionValidation.cs
// ===== RaghavStaadExtractor.ExecutionValidation =====

// Decompiled member-by-member from RaghavStaadExtractor.dll



private readonly string aesKey;


private readonly string aesIV;


private readonly string[] fakeErrors;


public ExecutionValidation()
{
	aesKey = "AshuVedantRaghav";
	aesIV = "AshuVedantRaghav";
	fakeErrors = new string[10] { "Runtime Error 0x80070005: Access is denied.", "Unhandled Exception: System.IO.FileLoadException", "Fatal Error: Missing dependency 'mscorlib.dll'.", "Memory read violation at address 0x00000014.", "System.NullReferenceException at MainModule()", "Unexpected token in config file at line 42.", "Access violation while reading registry key.", "System.BadImageFormatException: Invalid executable format.", "Fatal: CLR runtime initialization failed.", "Exception code 0xc0000005: Access violation." };
}


using System;
using System.IO;
using Microsoft.VisualBasic.CompilerServices;

internal bool IsLicenceValid()
{
	try
	{
		DateTime now = DateTime.Now;
		DateTime result = DateTime.MinValue;
		string path = "C:\\PlateNova\\PlateNova.tst";
		if (File.Exists(path))
		{
			string encryptedBase = File.ReadAllText(path);
			string s = AES_Decrypt(encryptedBase);
			if (!DateTime.TryParse(s, out result))
			{
				ShowFakeError();
				return false;
			}
			if (DateTime.Compare(now, result) < 0)
			{
				ShowFakeError();
				return false;
			}
		}
		string text = GetBiosSerial().ToUpper();
		if (string.IsNullOrWhiteSpace(text))
		{
			ShowFakeError();
			return false;
		}
		bool flag = false;
		DateTime t = new DateTime(2025, 12, 31);
		if (DateTime.Compare(DateTime.Today, t) <= 0)
		{
			flag = true;
		}
		else
		{
			string text2 = "C:\\PlateNova\\ExecutionFile.lic";
			if (!File.Exists(text2))
			{
				ShowFakeError();
				return false;
			}
			LicenseCheckResult licenseCheckResult = ParseLicenseFile(text2, text);
			if (licenseCheckResult.Status != 0)
			{
				ShowFakeError();
				return false;
			}
			flag = true;
		}
		if (flag)
		{
			try
			{
				string plainText = now.ToString("yyyy-MM-dd HH:mm:ss");
				string text3 = AES_Encrypt(plainText);
				if (!string.IsNullOrEmpty(text3))
				{
					File.WriteAllText(path, text3);
				}
			}
			catch (Exception projectError)
			{
				ProjectData.SetProjectError(projectError);
				ProjectData.ClearProjectError();
			}
			return true;
		}
		return false;
	}
	catch (Exception projectError2)
	{
		ProjectData.SetProjectError(projectError2);
		ShowFakeError();
		bool result2 = false;
		ProjectData.ClearProjectError();
		return result2;
	}
}


using System;
using System.IO;
using Microsoft.VisualBasic.CompilerServices;

private LicenseCheckResult ParseLicenseFile(string licPath, string biosSerial)
{
	try
	{
		string[] array = File.ReadAllLines(licPath);
		string[] array2 = array;
		foreach (string encryptedBase in array2)
		{
			string text = AES_Decrypt(encryptedBase).Trim();
			if (string.IsNullOrWhiteSpace(text) || text.StartsWith("#") || text.StartsWith("//") || !text.Contains("="))
			{
				continue;
			}
			string[] array3 = text.Split('=');
			if (array3.Length != 2)
			{
				continue;
			}
			string right = array3[0].Trim().ToUpper();
			string text2 = array3[1].Trim().ToLower();
			if (Operators.CompareString(biosSerial, right, TextCompare: false) != 0)
			{
				continue;
			}
			if ((Operators.CompareString(text2, "forever", TextCompare: false) == 0) | (Operators.CompareString(text2, "permanent", TextCompare: false) == 0))
			{
				return new LicenseCheckResult
				{
					Status = LicenseStatus.Valid,
					ExpiryDate = "Forever"
				};
			}
			if (DateTime.TryParse(text2, out var result))
			{
				if (DateTime.Compare(DateTime.Today, result) <= 0)
				{
					return new LicenseCheckResult
					{
						Status = LicenseStatus.Valid,
						ExpiryDate = result.ToString("yyyy-MM-dd")
					};
				}
				return new LicenseCheckResult
				{
					Status = LicenseStatus.Expired,
					ExpiryDate = result.ToString("yyyy-MM-dd")
				};
			}
			return new LicenseCheckResult
			{
				Status = LicenseStatus.InvalidFormat,
				ExpiryDate = text2
			};
		}
		return new LicenseCheckResult
		{
			Status = LicenseStatus.Unauthorized,
			ExpiryDate = ""
		};
	}
	catch (Exception ex)
	{
		ProjectData.SetProjectError(ex);
		Exception ex2 = ex;
		LicenseCheckResult result2 = new LicenseCheckResult
		{
			Status = LicenseStatus.InvalidFormat,
			ExpiryDate = ex2.Message
		};
		ProjectData.ClearProjectError();
		return result2;
	}
}


using System;
using System.IO;
using System.Security.Cryptography;
using System.Text;
using Microsoft.VisualBasic.CompilerServices;

private string AES_Decrypt(string encryptedBase64)
{
	try
	{
		byte[] bytes = Encoding.UTF8.GetBytes(aesKey);
		byte[] bytes2 = Encoding.UTF8.GetBytes(aesIV);
		byte[] buffer = Convert.FromBase64String(encryptedBase64);
		using Aes aes = Aes.Create();
		aes.Key = bytes;
		aes.IV = bytes2;
		aes.Padding = PaddingMode.PKCS7;
		using ICryptoTransform transform = aes.CreateDecryptor();
		using MemoryStream stream = new MemoryStream(buffer);
		using CryptoStream stream2 = new CryptoStream(stream, transform, CryptoStreamMode.Read);
		using StreamReader streamReader = new StreamReader(stream2);
		return streamReader.ReadToEnd();
	}
	catch (Exception projectError)
	{
		ProjectData.SetProjectError(projectError);
		string result = "";
		ProjectData.ClearProjectError();
		return result;
	}
}


using System;
using System.IO;
using System.Security.Cryptography;
using System.Text;
using Microsoft.VisualBasic.CompilerServices;

private string AES_Encrypt(string plainText)
{
	try
	{
		byte[] bytes = Encoding.UTF8.GetBytes(aesKey);
		byte[] bytes2 = Encoding.UTF8.GetBytes(aesIV);
		using Aes aes = Aes.Create();
		aes.Key = bytes;
		aes.IV = bytes2;
		aes.Padding = PaddingMode.PKCS7;
		using ICryptoTransform transform = aes.CreateEncryptor();
		using MemoryStream memoryStream = new MemoryStream();
		using (CryptoStream stream = new CryptoStream(memoryStream, transform, CryptoStreamMode.Write))
		{
			using StreamWriter streamWriter = new StreamWriter(stream);
			streamWriter.Write(plainText);
		}
		return Convert.ToBase64String(memoryStream.ToArray());
	}
	catch (Exception projectError)
	{
		ProjectData.SetProjectError(projectError);
		string result = "";
		ProjectData.ClearProjectError();
		return result;
	}
}


using System;
using System.Management;
using Microsoft.VisualBasic.CompilerServices;

private string GetBiosSerial()
{
	try
	{
		ManagementObjectSearcher managementObjectSearcher = new ManagementObjectSearcher("SELECT SerialNumber FROM Win32_BIOS");
		foreach (ManagementObject item in managementObjectSearcher.Get())
		{
			string text = item["SerialNumber"].ToString().Trim();
			if (!string.IsNullOrWhiteSpace(text))
			{
				return text;
			}
		}
	}
	catch (Exception projectError)
	{
		ProjectData.SetProjectError(projectError);
		ProjectData.ClearProjectError();
	}
	return "";
}


using System;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

private void ShowFakeError()
{
	try
	{
		Random random = new Random();
		string prompt = fakeErrors[random.Next(fakeErrors.Length)];
		Interaction.MsgBox(prompt, MsgBoxStyle.Critical, "Fatal Error");
		try
		{
			object obj = null;
			try
			{
				obj = RuntimeHelpers.GetObjectValue(Marshal.GetActiveObject("Excel.Application"));
			}
			catch (Exception projectError)
			{
				ProjectData.SetProjectError(projectError);
				ProjectData.ClearProjectError();
			}
			if (obj != null)
			{
				object objectValue = RuntimeHelpers.GetObjectValue(NewLateBinding.LateGet(obj, null, "ActiveWorkbook", new object[0], null, null, null));
				if (objectValue != null)
				{
					NewLateBinding.LateCall(objectValue, null, "Close", new object[1] { false }, null, null, null, IgnoreReturn: true);
				}
				Marshal.ReleaseComObject(RuntimeHelpers.GetObjectValue(obj));
			}
		}
		catch (Exception projectError2)
		{
			ProjectData.SetProjectError(projectError2);
			ProjectData.ClearProjectError();
		}
	}
	catch (Exception projectError3)
	{
		ProjectData.SetProjectError(projectError3);
		ProjectData.ClearProjectError();
	}
}


internal string GetCurrentBiosSerial()
{
	return GetBiosSerial();
}


using System;
using System.IO;
using Microsoft.VisualBasic.CompilerServices;

internal bool ValidateLicenseFileFormat(string licPath)
{
	try
	{
		if (!File.Exists(licPath))
		{
			return false;
		}
		string[] array = File.ReadAllLines(licPath);
		string[] array2 = array;
		foreach (string encryptedBase in array2)
		{
			string text = AES_Decrypt(encryptedBase).Trim();
			if (!string.IsNullOrWhiteSpace(text) && !text.StartsWith("#") && text.Contains("="))
			{
				string[] array3 = text.Split('=');
				if (array3.Length == 2 && !string.IsNullOrWhiteSpace(array3[0]) && !string.IsNullOrWhiteSpace(array3[1]))
				{
					return true;
				}
			}
		}
		return false;
	}
	catch (Exception projectError)
	{
		ProjectData.SetProjectError(projectError);
		bool result = false;
		ProjectData.ClearProjectError();
		return result;
	}
}
